vmsync/vmsync.go

```diff
@@ -0,0 +1,163 @@
+package vmsync
+
+import (
+	"app/base/logging"
+	"app/base/models"
+	"app/base/utils"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+var (
+	logger    *logrus.Logger
+	BatchSize = utils.GetEnv("BATCH_SIZE", 5000)
+)
+
+func init() {
+	logLevel := utils.GetEnv("LOGGING_LEVEL", "INFO")
+	var err error
+	logger, err = logging.CreateLogger(logLevel)
+	if err != nil {
+		fmt.Println("Error setting up logger.")
+		os.Exit(1)
+	}
+	logger.SetFormatter(&logrus.TextFormatter{
+		FullTimestamp: true,
+	})
+}
+
+func syncCveMetadata() {
+	apiCveMap, err := getAPICves()
+
+	if err != nil {
+		logger.Fatalf("Unable to get CVEs from VMaaS: %s", err)
+	}
+	logger.Infof("CVEs in VMaaS: %d", len(apiCveMap))
+
+	toSyncCves := make([]models.Cve, 0, len(apiCveMap))
+	for cveName, apiCve := range apiCveMap {
+		var severity models.Severity
+		err := severity.Scan(apiCve.Impact)
+		if err != nil {
+			severity = models.NotSet
+		}
+
+		cvss2Score, _ := strconv.ParseFloat(apiCve.Cvss2Score, 32)
```

Comment:
Better check the err

Reply:
Yep, done.
If the error I'm setting `cvss2/3Score` to 0.0